bxserum/provider/grpc.go

```diff
@@ -63,10 +72,72 @@ func NewGRPCClientWithOpts(opts RPCOpts) (*GRPCClient, error) {
 	if err != nil {
 		return nil, err
 	}
-	return &GRPCClient{
-		apiClient:  pb.NewApiClient(conn),
-		privateKey: opts.PrivateKey,
-	}, nil
+	client := &GRPCClient{
+		apiClient:                     pb.NewApiClient(conn),
+		privateKey:                    opts.PrivateKey,
+		recentBlockHashExpiryDuration: opts.CacheBlockHash,
+	}
+	go func() {
```

Comment:
this code also needs to be in `provider/ws.go` –– probably would be a good idea to create a struct encapsulating this logic?

Reply:
It'd also be nice to have a boolean flag in RPCOpts for if the user wants to enable this. Can default to `true`

Reply:
1. will think how to properly generalise it
2. basically we can use duration 0 for that, but flag will work too